app/helpers/lesson_plans_helper.rb

```diff
@@ -1,2 +1,9 @@
 module LessonPlansHelper
+  def lesson_plan_share_url!(lesson_plan, ext = nil)
```

Comment:
What's the reason for writing our own route helpers? So we don't have to remember the `.key!` or alias that with `to_param`?

Reply:
I had originally overridden to_param for this purpose, but I reverted it for two reasons:
1. it was too easy to inadvertently refer to the lesson plan path for non-sharing reasons (eg the add/remove form on lessons) and create a link record. I think this would definitely be mitigated by #501 - it's a little weird that we're using the id in the path to refer to the current lesson plan, and we shouldn't really ever need to, since users should only be able to view/edit the current lesson or view a shared one at a share link.
2. Relatedly, I'm not sure it really makes sense to think of it as the path for a given lesson...it could be the path to a read-only clone of that lesson.

But, I'd definitely be open to it if we do #501

Reply:
Ok, makes sense! (2) is a good point, I see the use in having a different helper to call when we want to be clear that it's a permalink.

Can we add something to config/routes.rb so that the list of helpers in `rake routes` remains complete? As an added benefit that'll give us non-bang versions of the method (which, some say, need to exist whenever there is a bang version).